.github/workflows/linux_unit_tests.yaml

```
@@ -106,14 +168,15 @@ jobs:
     - name: Run shell tests
       env:
           COVERAGE: true
+          PYTEST_ADDOPTS: '-m "not maybeslow"'
```

This marker seems stronger than just "maybe" now that you've taken the time to add it to more tests. Do we want to rename it to something like "long-running" (or just "slow")? Also, I think there may be tiers of slowness within these tests -- perhaps:
- < 1 min
- 1-2 min
- 3+ min

What do you think? Or should we focus on only the "slow" vs "not slow" distinction first?

```suggestion
          PYTEST_ADDOPTS: '-m "not longrunning"'
```

---

Also, I don't know if we're using parallel pytest processes at all? I know I mention pants a lot, but this is one thing pants is *really* good at: automatically running python tests in parallel with per-file caching. See https://www.pantsbuild.org/docs/python-test-goal. We would not need to modify spack's deployment model or change any code whatsoever for this to work -- we just need to add the pants runner script and a `pants.toml` file, and pants will work without modifying any python code. Please let me know if this is something I can investigate.

---

> Or should we focus on only the "slow" vs "not slow" distinction first?

I would say so. Right now I think ~150 tests out of ~2700 are responsible for 50% of the testing time. I'm absolutely fine using "slow" instead of "maybeslow" if people prefer that.

---

> Also, I don't know if we're using parallel pytest processes at all?

No we don't. Some of our tests are not properly isolated (which is unfortunate, I know). Running in parallel is something I hope we'll be able to do, but I think it currently requires some effort to track the tests that can step on each other and fix them. Also, I tried to run tests in parallel in the past only via `pytest-xdist` so my experience refers to that.